### PR TITLE
build: migrate to setup.cfg

### LINKED ISSUE #184: Update supported Python versions
The list of supported Python versions is outdated.
https://github.com/ianhi/mpl-interactions/blob/b5b3b3d7db51c4dc6bb8e0e3c851ca13e8b2ec26/setup.py#L38-L43
This for instance has the following effect:
[Supported Python versions](https://pypi.org/project/mpl-interactions)

Some suggestions related to the setup:
- Use `setup.cfg`
- Use [setuptools-scm](https://pypi.org/project/setuptools-scm)
- Put `tests` under a different folder (`mpl_interactions/tests` is currently installed through pip as well)
- Put source code under `src` (see [pytest recommendation](https://docs.pytest.org/en/stable/goodpractices.html#tests-outside-application-code))
- Run pytest through GitHub Actions on all supported Python versions --> although... fair point 7f59c397
- Add above badge to README
- Add [`project_urls`](https://packaging.python.org/guides/distributing-packages-using-setuptools/#project-urls)

### PR DESCRIPTION
**Pull request by redeboer**
Closes #184

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    %(jupyter)s
+    %(test)s
+    pre-commit
+
+[options.packages.find]
+where = src
+
+[options.package_data]
+ampform =
+    py.typed
```
Comment (ianhi) [inline setup.cfg:83]: left over from one of your packages?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    scipy
+    xarray
+dev =
+    %(doc)s
+    %(jupyter)s
+    %(test)s
+    pre-commit
+
+[options.packages.find]
+where = src
```
Comment (ianhi) [inline setup.cfg:79]: i think this is also leftover?

File: setup.cfg (hunk 1 above)
Comment (redeboer) [reply setup.cfg:83]: Haha :sweat_smile: Copy-paste ninja

File: setup.cfg (hunk 2 above)
Comment (redeboer) [reply setup.cfg:79]: Actually weird that even works

Comment (ianhi) [reply setup.cfg]: hahaha looking at the tests I'm not sure it did

**Pull request merged**